Duplicati/Library/Main/Controller.cs

```diff
@@ -671,21 +671,18 @@ private void SetupCommonOptions(ISetCommonOptions result, ref string[] paths, re
             foreach (Library.Interface.IGenericModule m in DynamicLoader.GenericLoader.Modules)
                 m_options.LoadedModules.Add(new KeyValuePair<bool, Library.Interface.IGenericModule>(Array.IndexOf<string>(m_options.DisableModules, m.Key.ToLower()) < 0 && (m.LoadAsDefault || Array.IndexOf<string>(m_options.EnableModules, m.Key.ToLower()) >= 0), m));
 
-            var conopts = new Dictionary<string, string>(m_options.RawOptions);
+            // Make the filter read-n-write able in the generic modules
+            var pristinefilter = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+            m_options.RawOptions["filter"] = pristinefilter;
+            
             var qp = new Library.Utility.Uri(m_backend).QueryParameters;
             foreach(var k in qp.Keys)
-                conopts[(string)k] = qp[(string)k];
-
-            // Make the filter read-n-write able in the generic modules
-            var pristinefilter = conopts["filter"] = string.Join(System.IO.Path.PathSeparator.ToString(), FilterExpression.Serialize(filter));
+                m_options.RawOptions[(string)k] = qp[(string)k];

```

Comment:
I think this will mean that all connection options (i.e. those passed as the query string) will overwrite basic options, like:
```
duplicati_cli backup "xyz://test?dblocksize=1mb" /sourcedata --dblock-size=2mb
```

Is there a reason for doing that?